src/templates/next/components/native/Table/Table.tsx

```diff
@@ -6,16 +6,39 @@ import BaseParagraph from "../../internal/BaseParagraph"
 import { Caption } from "../../../typography/Caption"
 import Prose from "../Prose"
 
+// Determine if the table is larger than the parent container (usually the screen width)
 const getIsTableOverflowing = (tableRef: React.RefObject<HTMLTableElement>) => {
   const tableRight = tableRef.current?.getBoundingClientRect().right || 0
   const parentRight =
     tableRef.current?.parentElement?.getBoundingClientRect().right || 0
   return tableRight > parentRight
 }
 
+// Determine the extent in which the table should have a sticky first part
+const getStickyRowIndexes = (tableRows: TableProps["content"]) => {
+  let stickyRowIndexes: number[] = [...Array(tableRows.length).keys()]
+
+  tableRows
+    .map((row) => row.content[0])
+    .forEach((cell, index) => {
+      if (!stickyRowIndexes.includes(index)) {
+        return
+      }
+
+      if (cell.rowSpan) {
+        stickyRowIndexes = stickyRowIndexes.filter(
+          (value) => value <= index || value >= index + cell.rowSpan!,
+        )
+      }
+    })
```

Comment:
Could you add a comment explaining what this section is doing? It's a little complicated haha

Reply:
Added in 39b45ee6bd712996fa47a82207a8789dba0ce2b2!